### PR TITLE
Hotfix/json unit tests

### PR DESCRIPTION
**Pull request by zakhenry**

### CODE REVIEW

**Comment by zakhenry** [comment]
@artstorm this PR adds methods to actually integration test json requests properly (previously we were just doing form encoded requests, which is wrong, as angular doesn't send form requests.)

There is a PR waiting on the laravel framework (https://github.com/laravel/framework/pull/9904) but it might not be merged, so I'm implemented the methods in the `TestCase`.

Also I had to fix up a bunch of controller that were using the `$request->get()` method, which is incorrect - it should be `$request->input()`.

@Redjik for your info, I found that some of the `PUT` requests were expecting the request to be nested in a `data: {}` key for some reason. This has been fixed. All json requests expect the entity or collection to be at the top level of the request body json.

**Comment by zakhenry** [comment]
@artstorm can you please review this PR. It looks like I've broken the float validator, but I'll fix that when I get back to my desk

**Comment by artstorm** [comment]
@xiphiaz I'm checking now. I'll see if I see what's up with the float validator while at it.

File: api/tests/integration/ArticleTest.php [hunk 1]
```diff
@@ -522,9 +528,6 @@ public function shouldGetCommentsForArticleUsingWithNestedHeader()
         $client->api('discussions')->remove($discussion['Discussion']['DiscussionID']);
     }
 
-    /**
-     * @test
-     */
```
Comment (artstorm) [inline api/tests/integration/ArticleTest.php:527]: I'm guessing disabling this test wasn't intentional, so I'll re-enable it. Which should also bring code coverage back up.

Comment (zakhenry) [inline api/tests/integration/ArticleTest.php:527]: @artstorm oh right, yea I thought it was a trailing group statement. Actually can you please update the test names to be consistent with the rest of the test suite? i.e. starting with `test`. It doesn't make sense to have two patterns.

Comment (artstorm) [inline api/tests/integration/ArticleTest.php:527]: @xiphiaz Will do.

**Comment by artstorm** [comment]
@xiphiaz PR looks super, nice fixes. I checked the 9904 issue, agree, this should be in the framework core.

Anyway, will merge when the tests have passed.

**Pull request merged**